spec/services/memberships/revoke_service_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Memberships::RevokeService, type: :service do
+  subject(:revoke_service) { described_class.new(membership.user) }
+
+  let(:membership) { create(:membership) }
+
+  describe '#revoke' do
+    context 'when revoking my own membership' do
+      it 'returns an error' do
+        result = revoke_service.revoke(membership.id)
+
+        expect(result).not_to be_success
+        expect(result.error).to eq('revoke_own_membership')
+      end
+    end
+
+    context 'when membership is not found' do
+      it 'returns an error' do
+        result = revoke_service.revoke(nil)
+
+        expect(result).not_to be_success
+        expect(result.error).to eq('not_found')
+      end
+    end
+
+    context 'when revoking another membership' do
+      let(:another_membership) { create(:membership) }
+
+      it 'revokes the memembership' do
+        result = revoke_service.revoke(another_membership.id)
+
+        expect(result).to be_success
+        expect(result.membership.id).to eq(another_membership.id)
```

Comment:
Can we also add an assertion to ensure that the membership is correctly revoked?

Reply:
Done